File: Easy/746. Min Cost Climbing Stairs/746. Min Cost Climbing Stairs.py
from typing import List


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        if n == 2:
            return min(cost[0], cost[1])

        prev1 = cost[0]
        prev2 = cost[1]

        for i in range(2, n):
            current = cost[i] + min(prev1, prev2)
            prev1, prev2 = prev2, current

        return min(prev1, prev2)

File: Easy/746. Min Cost Climbing Stairs/746. Min Cost Climbing Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        if (n == 2) {
            return Math.min(cost[0], cost[1]);
        }
        
        int prev1 = cost[0];
        int prev2 = cost[1];
        
        for (int i = 2; i < n; i++) {
            int current = cost[i] + Math.min(prev1, prev2);
            prev1 = prev2;
            prev2 = current;
        }
        
        return Math.min(prev1, prev2);
    }
}
